social_protection/validation.py

```diff
@@ -11,59 +12,65 @@ class BenefitPlanValidation(BaseModelValidation):
     def validate_create(cls, user, **data):
-        incoming_code = data.get('code')
-        code_error = check_bf_unique_code(incoming_code)
-        if code_error:
-            raise ValidationError(code_error[0]['message'])
-        incoming_name = data.get('name')
-        name_error = check_bf_unique_name(incoming_name)
-        if name_error:
-            raise ValidationError(name_error[0]['message'])
-        incoming_schema = data.get('beneficiary_data_schema')
-        schema_error = is_valid_json_schema(incoming_schema)
-        if schema_error:
-            raise ValidationError(schema_error[0]['message'])
+        errors = validate_benefit_plan(data)
+        if errors:
+            raise ValidationError(errors)
         super().validate_create(user, **data)
 
     @classmethod
     def validate_update(cls, user, **data):
         uuid = data.get('id')
-        incoming_code = data.get('code')
-        code_error = check_bf_unique_code(incoming_code, uuid)
-        if code_error:
-            raise ValidationError(code_error[0]['message'])
-        incoming_name = data.get('name')
-        name_error = check_bf_unique_name(incoming_name, uuid)
-        if name_error:
-            raise ValidationError(name_error[0]['message'])
-        incoming_schema = data.get('beneficiary_data_schema')
-        schema_error = is_valid_json_schema(incoming_schema)
-        if schema_error:
-            raise ValidationError(schema_error[0]['message'])
+        errors = validate_benefit_plan(data, uuid)
+        if errors:
+            raise ValidationError(errors)
         super().validate_update(user, **data)
 
     @classmethod
     def validate_delete(cls, user, **data):
         super().validate_delete(user, **data)
 
 
-def check_bf_unique_code(code, uuid=None):
+def validate_benefit_plan(data, uuid=None):
+    return [
+        *validate_not_empty_field(data.get("code"), "code"),
+        *validate_bf_unique_code(data.get('code'), uuid),
+        *validate_not_empty_field(data.get("name"), "name"),
+        *validate_bf_unique_name(data.get('name'), uuid),
+        *validate_json_schema(data.get('beneficiary_data_schema'))
+    ]
+
+
+def validate_bf_unique_code(code, uuid=None):
     instance = BenefitPlan.objects.filter(code=code, is_deleted=False).first()
     if instance and instance.uuid != uuid:
-        return [{"message": "BenefitPlan code %s already exists" % code}]
+        return [{"message": _("social_protection.validation.benefit_plan.code_exists" % {
+            'code': code
+        })}]
     return []
 
 
-def check_bf_unique_name(name, uuid=None):
+def validate_bf_unique_name(name, uuid=None):
     instance = BenefitPlan.objects.filter(name=name, is_deleted=False).first()
     if instance and instance.uuid != uuid:
-        return [{"message": "BenefitPlan name %s already exists" % name}]
+        return [{"message": _("social_protection.validation.benefit_plan.name_exists" % {
+            'name': name
+        })}]
```

Comment:
When you're using django translations translations I think that it's necessary (or at least reccomended to put a placeholder in string when you're providing parameters as attributes. For instance: instead of ` msgstr "BenefitPlan code %s already exists` it should be ` msgstr "BenefitPlan code %(code) already exists`

Reply:
done